src/services/__tests__/cgnOperatorSearchService.test.ts

```diff
@@ -0,0 +1,378 @@
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should make the correct api call", async () => {
+    const service = new CgnOperatorSearchService(api);
+
+    await service.getMerchant(aMerchantId);
+
+    expect(mockGetMerchant).toHaveBeenCalledWith({
+      merchantId: aMerchantId
+    });
+  });
+
+  it("should handle a success response", async () => {
+    const service = new CgnOperatorSearchService(api);
+
+    const res = await service.getMerchant(aMerchantId);
+
+    expect(res).toMatchObject({
+      kind: "IResponseSuccessJson"
+    });
+
+    // @ts-ignore
+    expect(res.value).toMatchObject(anExpectedResponse);
+  });
+
+  it("should handle a not found error when the CGN is not found", async () => {
+    mockGetMerchant.mockImplementationOnce(() => t.success({ status: 404 }));
+
+    const service = new CgnOperatorSearchService(api);
+
+    const res = await service.getMerchant(aMerchantId);
+
+    expect(res).toMatchObject({
+      kind: "IResponseErrorNotFound"
+    });
+  });
+
+  it("should handle an internal error response", async () => {
+    const aGenericProblem = {};
+    mockGetMerchant.mockImplementationOnce(() =>
+      t.success({ status: 500, value: aGenericProblem })
+    );
+
+    const service = new CgnOperatorSearchService(api);
+
+    const res = await service.getMerchant(aMerchantId);
+
+    expect(res).toMatchObject({
+      kind: "IResponseErrorInternal"
+    });
+  });
+
+  it("should return an error for unhandled response status code", async () => {
+    mockGetMerchant.mockImplementationOnce(() => t.success({ status: 123 }));
+    const service = new CgnOperatorSearchService(api);
+
+    const res = await service.getMerchant(aMerchantId);
+
+    expect(res).toMatchObject({
+      kind: "IResponseErrorInternal"
+    });
+  });
+
+  it("should return an error if the api call thows", async () => {
+    mockGetMerchant.mockImplementationOnce(() => {
+      throw new Error();
+    });
+    const service = new CgnOperatorSearchService(api);
+
+    const res = await service.getMerchant(aMerchantId);
+
+    expect(res).toMatchObject({
+      kind: "IResponseErrorInternal"
+    });
+  });
+});
+describe("CgnOperatorSearchService#getOnlineMerchants", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should make the correct api call", async () => {
+    const service = new CgnOperatorSearchService(api);
+
+    await service.getOnlineMerchants(anOnlineMerchantSearchRequest);
+
+    expect(mockGetOnlineMerchants).toHaveBeenCalledWith({
+      body: anOnlineMerchantSearchRequest
+    });
+  });
+
+  it("should handle a success response", async () => {
+    const service = new CgnOperatorSearchService(api);
+
+    const res = await service.getOnlineMerchants(anOnlineMerchantSearchRequest);
+
+    expect(res).toMatchObject({
+      kind: "IResponseSuccessJson"
+    });
+
+    // @ts-ignore
+    expect(res.value).toMatchObject(anApiResult);
```

Comment:
Use `@ts-ignore` only if you want to test code typings.
You can use this flavor instead
 ```suggestion
    expect(res.kind === "IResponseSuccessJson" && res.value).toMatchObject(anApiResult);
```